### DISCUSSION TITLE
“RuntimeError: Timeout context manager should be used inside a task” when using asyncio

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18335] [Q&A] [answered]
Has anyone run into `RuntimeError: Timeout context manager should be used inside a task` when running asyncio? I'm using asyncio.gather on two calls: one with aiohttp and one with asyncpg.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701192009821919?thread_ts=1701192009.821919&cid=C01U954MEER).*

**Comment by smackesey** [comment]
Where are you running into this in the context of Dagster?

**Reply by ryanruaneyougov** [reply, marked as answer]
I have been looking into it and I think I see the problem. While I'm using a context manager for the aiohttp `session.post`, I believe I am improperly storing the aiohttp session itself in a Resource. This though raises the question of how to handle async pools/sessions inside Dagster resources, for which a discussion has been raised: https://github.com/dagster-io/dagster/discussions/18159
